### PR TITLE
Horovod+tf2.2 compatibility

### LINKED ISSUE #1688: TensorFlow master branch (tf-nightly) compatibility
See: https://github.com/tensorflow/tensorflow/issues/35138

In TensorFlow 2.1, `experimental_run_tf_function` was removed from `tf.keras.Model.compile` at this commit: c73c99c#diff-de9b96ac2d81503324cbbbe21732031fR1159

In Horovod, this flag / graph mode is necessary in order for `Optimizer.get_gradients()` to be called, which aggregates gradients across workers. Since this flag has been removed, distributed training in Horovod with tf.keras isn't working in the example scripts for TensorFlow 2.1.

@abditag2 is currently investigating a workaround.  Note that we cannot perform the allreduce aggregation in apply_gradients due to interactions with gradient clipping and loss scaling (see horovod/horovod#1347).

### PR DESCRIPTION
**Pull request by abditag2**
Fixes #1688.

### CODE REVIEW

File: horovod/_keras/__init__.py [hunk 1]
```diff
@@ -13,22 +13,76 @@
...
+            Unscales the gradients by the loss scale.
 
+            See LossScaleOptimizer.get_unscaled_gradients() for more info.
+
+            In DistributedOptimizer, get_unscaled_gradients() is overriden to
+            allreduce the gradients before returning them.
+            """
+            self._get_gradient_or_get_unscaled_gradients_called = True
+            gradients = super(self.__class__, self).get_unscaled_gradients(*args, **kwargs)
+            return _allreduce_gradients(gradients, self._name, self._sparse_as_dense,
```
Comment (tgaddair) [inline horovod/_keras/__init__.py]: I believe we would need to do the allreduce before getting the unscaled gradients to be consistent with past behavior.  @DEKHTIARJonathan can confirm.

**Comment by tgaddair** [comment]
See: https://github.com/tensorflow/tensorflow/issues/36398

cc: @romerojosh @DEKHTIARJonathan

File: Dockerfile.test.cpu [hunk 2]
```diff
@@ -97,7 +97,7 @@ RUN pip install ${TENSORFLOW_PACKAGE}
 
 # Install Keras.
 # Pin scipy<1.4.0: https://github.com/scipy/scipy/issues/11237
-RUN pip install ${KERAS_PACKAGE} h5py "scipy<1.4.0" pandas
+RUN pip install ${KERAS_PACKAGE} h5py "scipy<1.4.0" pandas==0.24.2
```
Comment (abditag2) [inline Dockerfile.test.cpu]: will be removed after petastorm update.

**Comment by tgaddair** [comment]
Fixed by #1777.
